### PR TITLE
docs:#888 explain versioning

### LINKED ISSUE #888: Create a page to explain what is Redpanda's versioning scheme
### Describe the Issue

During the execution of this [PR](https://github.com/redpanda-data/documentation/pull/859) we mention that users should only upgrade one major version at a time. But we never clearly inform what constitutes a major version for us. 

So, ideally we should create a page to clearly explain to our users what is our versioning scheme. Essentially this page should have:

- Why do we number our versions like that.
- What do we define as a major version release.
- What is a minor version.
- Why we don't have specific docs for minor versions.
- And any other subjects relative to versions, like for example, if we respect common versioning schemes like [SemVer](https://semver.org/)

### New feature or content gap requests

- Gap

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #888 

This adds info in our Intro to Redpanda page about our versioning. Specific details about upgrade/downgrade were added in #1246 to the [Upgrade pages](https://docs.redpanda.com/docs/manage/cluster-maintenance/rolling-upgrade/). This just calls out a new **How to get started section** in the intro with general versioning info. (The original issue asked for explanation of major vs minor version, but we don't use that terminology in our docs anymore.) 

Preview: https://deploy-preview-1498--redpanda-documentation.netlify.app/docs/get-started/intro-to-events/#how-to-get-started

### CODE REVIEW

**Comment by micheleRP** [comment]
Hi @bpraseed. This adds the line, "Redpanda Cloud releases on a continuous basis and uptakes Redpanda platform versions." Please let me know if this line is correct, should be rephrased to something else, or removed altogether!

File: docs/get-started/intro-to-events.mdx [hunk 1]
```diff
@@ -59,13 +59,20 @@ Redpanda is less complex and less costly than any other commercial mission-criti
 
   With the [Bring Your Own Cloud](../../deploy/deployment-option/cloud/dedicated-byoc) (BYOC) offering, you deploy Redpanda in your own virtual private cloud, and all data is contained in your environment. Redpanda handles provisioning, monitoring, and upgrades, but you manage your streaming data without Redpanda’s control plane ever seeing it.
 
-## Next steps
-To learn more about Redpanda, see [How Redpanda Works](../architecture).
+## How to get started
 
 You can deploy Redpanda in a self-hosted environment (Redpanda platform) or as a fully-managed cloud service (Redpanda Cloud). 
+
+Redpanda platform version numbers follow the convention AB.C.D, where AB is the two digit year, C is the feature release, and D is the patch release. For example, version 22.3.1 indicates the first patch release on the third feature release of the year 2022. Patch releases include bug fixes and minor improvements, with no change to user-facing behavior. New and enhanced features are documented with each feature release.
```
Comment (JakeSCahill) [inline docs/get-started/intro-to-events.mdx:66]: This is different from our [upgrade doc](https://docs.redpanda.com/docs/manage/cluster-maintenance/rolling-upgrade/). The only difference is that we say Redpanda platform here, but we don't explain what Redpanda platform is compared to plain old Redpanda.

File: docs/get-started/intro-to-events.mdx [hunk 2]
```diff
@@ -59,13 +59,20 @@ Redpanda is less complex and less costly than any other commercial mission-criti
...
-To learn more about Redpanda, see [How Redpanda Works](../architecture).
+## How to get started
 
 You can deploy Redpanda in a self-hosted environment (Redpanda platform) or as a fully-managed cloud service (Redpanda Cloud). 
+
+Redpanda platform version numbers follow the convention AB.C.D, where AB is the two digit year, C is the feature release, and D is the patch release. For example, version 22.3.1 indicates the first patch release on the third feature release of the year 2022. Patch releases include bug fixes and minor improvements, with no change to user-facing behavior. New and enhanced features are documented with each feature release. 
+
+Redpanda Cloud releases on a continuous basis and uptakes Redpanda platform versions. 
+
 - To get started with a self-hosted Redpanda platform deployment, see [Redpanda Licensing](../licenses).
```
Comment (JakeSCahill) [inline docs/get-started/intro-to-events.mdx:70]: Is the licensing doc where we expect users to get started? Or our quickstart?

File: docs/get-started/intro-to-events.mdx [hunk 3]
```diff
@@ -59,13 +59,20 @@ Redpanda is less complex and less costly than any other commercial mission-criti
...
 - To get started with a Redpanda Cloud deployment, see [Dedicated Cloud vs BYOC](../../deploy/deployment-option/cloud/dedicated-byoc/).
```
Comment (JakeSCahill) [inline docs/get-started/intro-to-events.mdx:71]: Same comment as above. Is the quickstart a better place?

File: docs/get-started/intro-to-events.mdx (hunk 1 above)
Comment (micheleRP) [reply docs/get-started/intro-to-events.mdx:66]: Thanks @JakeSCahill! I updated the upgrade pages with the same info.

File: docs/get-started/intro-to-events.mdx (hunk 3 above)
Comment (micheleRP) [reply docs/get-started/intro-to-events.mdx:71]: Yes, good catch. I moved these links to the Next steps section and added a link there for the quickstart. And then I changed the title of this heading to Redpanda versions, so it'll be easier to search/find (which was goal of this issue).

File: docs/get-started/intro-to-events.mdx [hunk 4]
```diff
@@ -59,13 +59,20 @@ Redpanda is less complex and less costly than any other commercial mission-criti
 
   With the [Bring Your Own Cloud](../../deploy/deployment-option/cloud/dedicated-byoc) (BYOC) offering, you deploy Redpanda in your own virtual private cloud, and all data is contained in your environment. Redpanda handles provisioning, monitoring, and upgrades, but you manage your streaming data without Redpanda’s control plane ever seeing it.
 
-## Next steps
-To learn more about Redpanda, see [How Redpanda Works](../architecture).
+## How to get started
```
[the remark this answers is not in the document]
Comment (micheleRP) [reply docs/get-started/intro-to-events.mdx:62]: Yes, thanks @Deflaimun! I changed the title of this section to Redpanda platform versions.

**Comment by bpraseed** [comment]
Lgtm.

**Review by mattschumpert — APPROVED** [review]
LGTM

**Review by Feediver1 — APPROVED** [review]
lgtm

**Pull request merged**